calico_kubernetes/tests/kube_plugin_test.py

```diff
@@ -108,11 +116,104 @@ def test_delete(self):
 
             # Assert
             m_container_remove.assert_called_once_with()
-            self.assertEqual(namespace, self.plugin.namespace)
-            self.assertEqual(pod_name, self.plugin.pod_name)
-            self.assertEqual(docker_id, self.plugin.docker_id)
-            self.assertEqual(profile_name, self.plugin.profile_name)
-            m_datastore_client.remove_profile(profile_name)
+            assert_equal(namespace, self.plugin.namespace)
+            assert_equal(pod_name, self.plugin.pod_name)
+            assert_equal(docker_id, self.plugin.docker_id)
+            assert_equal(profile_name, self.plugin.profile_name)
+            self.m_datastore_client.remove_profile(profile_name)
+
+    def test_delete_error(self):
+        """Test Pod Deletion Hook Failure"""
+        """
+        If the datastore remove_profile function returns KeyError, the profile is not in the datastore.
+        Issue warning log, but do not fail.
+        """
+        with patch_object(self.plugin, '_container_remove', autospec=True) as m_container_remove:
+            # Set up mock objs
+            self.m_datastore_client.profile_exists.return_value = True
+            self.m_datastore_client.remove_profile.side_effect = KeyError
+
+            # Set up args
+            namespace = 'ns'
+            pod_name = 'pod1'
+            docker_id = 123456789101112
+            profile_name = 'ns_pod1_123456789101'
+
+            # Call method under test
+            self.plugin.delete(namespace, pod_name, docker_id)
```

Comment:
Much neater -- but I think you should add a comment trailing this function saying something like "We don't need to assert anything here; if WAD we won't raise an exception."

It's a fine optimization to make (which is why I suggested it), but folk reading this code later will wonder if you just forgot to write asserts unless you call out why you omitted them.